QuizApp/App/Screens/Quiz/QuizListViewModel.swift

```diff
@@ -14,14 +15,42 @@ final class QuizListViewModel {
 
 extension QuizListViewModel {
 
-    func fetchQuiz() {
+    func fetchAllQuizes() {
         Task(priority: .background) {
             let quizes = try await quizUseCase.fetchQuizes()
 
             DispatchQueue.main.async { [weak self] in
                 guard let self = self else { return }
                 self.quizes = quizes
                     .map { QuizViewModel(from: $0) }
+                self.categories = quizes
+                    .map { CategoryViewModel(from: $0.category) }
+//                    .unique()
+            }
+        }
+    }
+
+    func fetchQuiz(for category: String) {
+        Task(priority: .background) {
+            let quizes = try await quizUseCase.fetchQuizes(for: category)
+
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+                self.quizes = quizes
+                    .map { QuizViewModel(from: $0) }
+            }
+        }
+    }
+
+    func fetchCategories() {
```

Comment:
You are already assigning categories when you call fetchAllQuizzes(), so you don't need this.